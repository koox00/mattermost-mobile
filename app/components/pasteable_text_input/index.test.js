// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {shallow} from 'enzyme';
import React from 'react';
import {NativeEventEmitter} from 'react-native';

import {PASTE_FILES} from '@constants/post_draft';
import EventEmitter from '@mm-redux/utils/event_emitter';

import {PasteableTextInput} from './index';

const nativeEventEmitter = new NativeEventEmitter();

describe('PasteableTextInput', () => {
    const emit = jest.spyOn(EventEmitter, 'emit');

    test('should render pasteable text input', () => {
        const onPaste = jest.fn();
        const text = 'My Text';
        const component = shallow(
            <PasteableTextInput
                onPaste={onPaste}
                screenId='Channel'
            >{text}</PasteableTextInput>,
        );
        expect(component).toMatchSnapshot();
    });

    test('should call onPaste props if native onPaste trigger', () => {
        const event = {someData: 'data'};
        const text = 'My Text';
        shallow(
            <PasteableTextInput screenId='Channel'>{text}</PasteableTextInput>,
        );
        nativeEventEmitter.emit('onPaste', event);
        expect(emit).toHaveBeenCalledWith(PASTE_FILES, null, event, 'Channel');
    });

    test('should remove onPaste listener when unmount', () => {
        const mockRemove = jest.fn();
        const text = 'My Text';
        const component = shallow(
            <PasteableTextInput screenId='Channel'>{text}</PasteableTextInput>,
        );

        component.instance().subscription.remove = mockRemove;
        component.instance().componentWillUnmount();
        expect(mockRemove).toHaveBeenCalled();
    });

    test('should emit PASTE_FILES event only for last subscription', () => {
        const component1 = shallow(<PasteableTextInput screenId='Channel'/>);
        const instance1 = component1.instance();
        const component2 = shallow(<PasteableTextInput screenId='Thread'/>);
        const instance2 = component2.instance();

        instance1.onPaste();
        expect(emit).not.toHaveBeenCalled();
        instance2.onPaste();
        expect(emit).toHaveBeenCalledTimes(1);
    });
});
