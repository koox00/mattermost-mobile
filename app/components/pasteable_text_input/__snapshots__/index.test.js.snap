// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasteableTextInput should render pasteable text input 1`] = `
<TextInput
  allowFontScaling={true}
  onPaste={[Function]}
  rejectResponderTermination={true}
  screenId="Channel"
  underlineColorAndroid="transparent"
>
  My Text
</TextInput>
`;
